packages/block-editor/src/components/link-control/index.js

```diff
@@ -136,6 +138,15 @@ function LinkControl( {
 	const textInputRef = useRef();
 	const isEndingEditWithFocus = useRef( false );
 
+	const showIconLabels = useSelect(
+		( select ) =>
+			select( preferencesStore ).get(
+				'core/edit-post',
```

Comment:
This is a block editor component, so it gets used across both post and site editors. Unfortunately the text label settings are specific to each editor, so using `edit-post` preferences here means that if the preference is set differently in the site editor, this component won't reflect it there.

Reply:
Great spot @tellthemachines. Some more context on @abhi3315. Basically the block editor package is designed to be used standalone outside of "WordPress". Therefore we can't query anything like `core` store or even preferences namespaces.

To work around this we often create editor "settings" which we:

- set in the editor instances (`post`, `widgets` and `site`) - passing through to the block editor component
- consume within the block editor package (e.g. here in Link Control) to access the data

 And example of this can be found in
 
 `packages/edit-site/src/components/block-editor/index.js`

If you do it like this you could have a single setting for the text labels passed into the block editor. Then you could consume that value.

Does that sound suitable @tellthemachines?

Reply:
In this case we don't need to create an editor setting, because the classname we hook the styles onto is being applied to the interface component in both the post and the site editor, so we can make this change just by adding some CSS ✨